### ISSUE TITLE
Allow clone repositories tool to specify clone depth

### ISSUE THREAD

**Issue by Al3nMicL** [microsoft/devhome#3607]
### Suggested new feature or improvement

First of all, this tool is pretty cool! Great work Windows Dev team... With that being said, I hope to see new features, such as the ability to specify a clone depth when cloning GitHub repos. 

### Scenario

For instance, I was trying to clone the Supabase/Supabase GitHub repo so I can try out my own self hosted instance using Podman. The instructions on their [self hosting setup page](https://supabase.com/docs/guides/self-hosting/docker) mention to pass the --depth 1 param when cloning, because the most recent commit is all that is needed for setup. Pulling the most recent commit from master yields about a 650 Mb download, as opposed to the ~2.6 Gb a clone of the entire master branch would be. This is a huge difference in disk usage from passing one parameter... And it can definitely add up if a user comes across this when working with multiple repositories that have similar requirements.

### Additional details

TL;DR: allow an advanced toggle in clone repo tool where users can specify clone depth.
Thanks :)
